src/main/java/org/spongepowered/common/event/SpongeEventManager.java

```diff
@@ -68,10 +70,8 @@
     private final Object lock = new Object();
 
     private final PluginManager pluginManager;
-    private final DefineableClassLoader classLoader = new DefineableClassLoader(getClass().getClassLoader());
-    private final AnnotatedEventListener.Factory handlerFactory = new ClassEventListenerFactory("org.spongepowered.common.event.listener",
-            new FilterFactory("org.spongepowered.common.event.filters", this.classLoader), this.classLoader);
     private final Multimap<Class<?>, RegisteredListener<?>> handlersByEvent = HashMultimap.create();
+    private final Map<ClassLoader, AnnotatedEventListener.Factory> classLoaders = Maps.newHashMap();
```

Comment:
Keeps us from leaking a plugin's special classloader if they've released it to be garbage collected. Which is pretty common as when the classloader is garbage collected so is all of the classes it loaded.